compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -305,6 +305,18 @@ classifyPrimitive(CallExpr *call) {
     return NOT_FAST_NOT_LOCAL;
 
   case PRIM_GPU_KERNEL_LAUNCH:
+  case PRIM_GPU_THREADIDX_X:
```

Comment:
FWIW all of these new intrinsics are in the FAST_AND_LOCAL for the purposes of this analysis. (Of course it's not so clear to me why it would matter in the near term, since we can't have GPU code without a kernel launch, which, is not "fast" i.e. able to run in an active message handler, but I'd like to keep this optimization accurate since we might evolve it for other uses).

Reply:
Thanks for letting me know. I added them there since that's where PRIM_GPU_KERNEL_LAUNCH was.

I've moved them to FAST_AND_LOCAL in 28c18684651d47325913f8b5fc090659d4276a6b

Could you give me a brief overview of what this analysis does. What does "fast and local mean?" so if I add new primitives I'll have a better idea of what to categorize them as.

Reply:
Sure. This analysis is for deciding whether or not an `on` statement can run in an active message handler. The handler can run things that don't communicate, don't block, and are short in duration. In the compiler/runtime we call `on` statements that will be run in the handler "fast" on statements. I don't remember the history - for the analysis specifically - for why it distinguishes between "fast" and "local" but I think it was just easier to combine the information that way & that the "fast on" optimization needs it to be both fast and local.